core/src/main/java/org/infinispan/marshall/core/internal/ExternalJBossMarshaller.java

```diff
@@ -0,0 +1,226 @@
+   }
+
+   @Override
+   public byte[] objectToByteBuffer(Object obj, int estimatedSize) throws IOException, InterruptedException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public byte[] objectToByteBuffer(Object obj) throws IOException, InterruptedException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public Object objectFromByteBuffer(byte[] buf) throws IOException, ClassNotFoundException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public Object objectFromByteBuffer(byte[] buf, int offset, int length) throws IOException, ClassNotFoundException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public ByteBuffer objectToBuffer(Object o) throws IOException, InterruptedException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public BufferSizePredictor getBufferSizePredictor(Object o) {
+      throw new RuntimeException("NYI");
+   }
+
+   static final class JBossByteOutput extends OutputStream implements ByteOutput {
+
+      final ObjectOutput out;
+
+      JBossByteOutput(ObjectOutput out) {
+         this.out = out;
+      }
+
+      @Override
+      public void write(int b) throws IOException {
+         out.write(b);
+      }
+
+   }
+
+   static final class JBossByteInput extends InputStream implements ByteInput {
+
+      final ObjectInput in;
+
+      JBossByteInput(ObjectInput in) {
+         this.in = in;
+      }
+
+      @Override
+      public int read() throws IOException {
+         try {
+            return in.readUnsignedByte();
+         } catch (ArrayIndexOutOfBoundsException e) {
+            // When JBoss Marshalling starts reading a stream, it buffers
+            // the contents, so it'll try as read as much as it can, so it
+            // can end up reading too far, in which case launder the exception
+            // into an IOException so that reading can stop without causing
+            // issues.
+            throw new EOFException();
+         }
+      }
+
+   }
+
+   static final class InternalJBossMarshaller extends JBossMarshaller {
+      final InternalExternalizerTable externalizers;
+
+      InternalJBossMarshaller(InternalExternalizerTable externalizers, GlobalConfiguration globalCfg) {
+         super(null, globalCfg);
+         this.externalizers = externalizers;
+      }
+
+      @Override
+      public void start() {
+         super.start();
+         baseCfg.setObjectTable(new ObjectTable() {
+            @Override
+            public Writer getObjectWriter(Object object) throws IOException {
+               InternalExternalizerTable.MarshallableType type = externalizers.marshallable(object);
+               switch (type) {
+                  case INTERNAL:
+                  case PREDEFINED:
+                  case ANNOTATED:
+                     return (m, obj) -> {
+                        Externalizer<Object> ext = externalizers.findWriteExternalizer(obj, m);
```

Comment:
This is looking up the same key a second time in the same maps. I'd remove the `externalizers.marshallable(object)` call and call `findWriteExternalizer` directly.